app/controllers/candidate_interface/personal_statement_controller.rb

```diff
@@ -37,16 +40,16 @@ def update
       @becoming_a_teacher_form = BecomingATeacherForm.build_from_params(becoming_a_teacher_params)
       @return_to = return_to_after_edit(default: candidate_interface_becoming_a_teacher_show_path)
 
-      if @becoming_a_teacher_form.save(current_application)
-        if @becoming_a_teacher_form.blank?
-          set_section_to_incomplete_if_completed
-          redirect_to candidate_interface_application_form_path
-        else
-          redirect_to @return_to[:back_path]
-        end
+      @becoming_a_teacher_form.save(current_application)
+
+      if @becoming_a_teacher_form.invalid?
+        set_section_to_incomplete_if_completed
+      end
+
+      if @becoming_a_teacher_form.blank?
+        redirect_to candidate_interface_application_form_path
       else
-        track_validation_error(@becoming_a_teacher_form)
```

Comment:
Do we lose visibility of other validation errors if we remove this?

Reply:
Yeah we do. Arguably that’s ok here as this isn’t a traditional validation error as it’s non-blocking? But possibly it’s still useful to track? cc @MylesJarvis